src/meta/ps3_sgh_sgb.c

```diff
@@ -70,24 +70,23 @@ VGMSTREAM * init_vgmstream_ps3_sgdx(STREAMFILE *streamFile) {
 
 
     /* WAVE chunk (size 0x10 + files * 0x38 + optional padding) */
-    /*  the format reads chunks until header_size, but we only want WAVE in the first position meaning BGM */
-    if (read_32bitBE(0x10,streamHeader) != 0x57415645)  /* "WAVE" */
-        goto fail;
-    /* 0x04  SGX: unknown; SGD/SGH: chunk length */
-    /* 0x08  null */
-
-    /* validate multi-streams (usually only SE containers) */
-    {
-        int total_streams = read_32bitLE(0x1c,streamHeader);
-        VGM_ASSERT(total_streams > 1, "SGXD: multiple streams found (%i entries)\n", total_streams);
-        if (target_stream == 0) target_stream = 1; /* auto: default to 1 */
-        if (target_stream > total_streams) goto fail;
+    /* 0x04  SGX: unknown; SGD/SGH: chunk length,  0x08  null */
+    if (is_sgx) { /* position after chunk+size */
+        if (read_32bitBE(0x10,streamHeader) != 0x57415645) goto fail;  /* "WAVE" */
+        chunk_offset = 0x18;
+    } else {
+        if (!find_chunk_le(streamHeader, 0x57415645,0x10,0, &chunk_offset,NULL)) goto fail; /* "WAVE" */
     }
 
+    /* check multi-streams (usually only SE containers; Puppeteer) */
+    total_streams = read_32bitLE(chunk_offset+0x04,streamHeader);
+    if (target_stream == 0) target_stream = 1;

```

Comment:
That's, uhm, a bit vague.

If you ask about the chunk_offset, it points after the id+size = 4+4 (makes more sense to me; say you have find_weird_chunk with id+size = 2+variable, you want it to return an offset to the chunk data, after the variable size)

If you ask about total/target_stream, if I implement a stream selector for multi-stream formats later, I'd think 1=first stream, N=Nth, while 0=none selected. Which now it means first stream, but I could make it give an error when none selected.

Otherwise, all sample files I have decode the same.